### PR TITLE
Add Pkl 0.26 features

### PR DESCRIPTION
**Pull request by holzensp**

### CODE REVIEW

**Review by bioball — CHANGES_REQUESTED** [review]
Looking pretty good! Left some comments.

Also: I'm seeing some formatting issues. Can you run `swiftformat` on these changes?

File: Tests/PklSwiftTests/EvaluatorTest.swift [hunk 1]
```diff
@@ -91,6 +91,33 @@ final class PklSwiftTests: XCTestCase {
...
+        let version = try getVersion()
+        let expected = version < pklVersion0_26
+                ? "http options are not supported on Pkl versions lower than 0.26"
+                : "ConnectException: Error connecting to host `example.com`"
+        var options = EvaluatorOptions.preconfigured
+        options.http = Http(
+                caCertificates: nil,
+                proxy: Proxy(
+                        address: "http://my.proxy.example.com:5080",
+                        noProxy: ["myhost.com:1337", "myotherhost.org:42"]))
```
[the remark this answers is not in the document]
Comment (stackoverflow) [reply Tests/PklSwiftTests/EvaluatorTest.swift:109]: We should pick an open port, otherwise this may lead to "sometimes works on my machine".

**Review by stackoverflow — APPROVED** [review]
LGTM

File: Sources/PklSwift/EvaluatorManager.swift [hunk 2]
```diff
@@ -37,6 +38,57 @@ public func withEvaluatorManager<T>(_ action: (EvaluatorManager) async throws ->
...
+    }
+    throw PklError("Unable to find `pkl` command on PATH.")
+}
+
+/// Get the version of the `pkl-server` being used.
+func getVersion() throws -> SemanticVersion {
+    let pklCommand = try getPklCommand()
+    let process = Process()
+    process.executableURL = URL(fileURLWithPath: pklCommand[0])
+    process.arguments = Array(pklCommand.dropFirst()) + ["--version"]
```
[the remark this answers is not in the document]
Comment (bioball) [reply Sources/PklSwift/EvaluatorManager.swift:76]: Oh, actually--bad suggestion, my bad.

File: Sources/PklSwift/EvaluatorManager.swift [hunk 3]
```diff
@@ -287,3 +344,11 @@ enum PklBugError: Error {
     case invalidEvaluatorId(String)
     case unknownMessage(String)
 }
+
+let pklVersion0_25 = SemanticVersion("0.25.0")!
+let pklVersion0_26 = SemanticVersion("0.26.0")!
+
+let supportedPklVersions = [
```
[the remark this answers is not in the document]
Comment (holzensp) [reply Sources/PklSwift/EvaluatorManager.swift:351]: Future proofing; having the tests automatically cover all supported cases. Happy to put this in the test, but I thought putting it next to the definitions of `pklVersion0_xx` would be safer.

File: Sources/PklSwift/EvaluatorOptions.swift [hunk 4]
```diff
@@ -227,28 +236,15 @@ extension EvaluatorOptions {
...
-        if evaluatorSettings.rootDir != nil {
-            options.rootDir = evaluatorSettings.rootDir
-        }
+        options.properties = evaluatorSettings.externalProperties ?? properties
+        options.env = evaluatorSettings.env ?? env
+        options.allowedModules = evaluatorSettings.allowedModules ?? allowedModules
+        options.allowedResources = evaluatorSettings.allowedResources ?? allowedResources
+        options.cacheDir = evaluatorSettings.noCache != nil ? nil : (evaluatorSettings.moduleCacheDir ?? cacheDir)
+        options.rootDir = evaluatorSettings.rootDir ?? rootDir
+        options.http = evaluatorSettings.http ?? http
```
[the remark this answers is not in the document]
Comment (holzensp) [reply Sources/PklSwift/EvaluatorOptions.swift]: Have you found ways to mute IDE warnings? Having dozens of "'self' is unnecessary" warnings makes the code analysis borderline useless.

File: Tests/PklSwiftTests/EvaluatorTest.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (holzensp) [reply Tests/PklSwiftTests/EvaluatorTest.swift:109]: Picking an open port leads to potential races :( There is no real "idea" solution here, I think. What about a reserved port that is not in use anymore (like `1`) or one that _should_ not be in use anymore (like `23`)?

File: Sources/PklSwift/EvaluatorManager.swift [hunk 5]
```diff
@@ -37,6 +38,57 @@ public func withEvaluatorManager<T>(_ action: (EvaluatorManager) async throws ->
...
+    guard let outputData = try pipe.fileHandleForReading.readToEnd(),
+          let output = String(data: outputData, encoding: .ascii)?.split(separator: " "),
+          output.count > 2,
+          output[0] == "Pkl"
+    else {
+        throw PklError("Could not get version from Pkl binary")
+    }
+
+    return SemanticVersion(String(output[1]))!
+}
```
[the remark this answers is not in the document]
Comment (holzensp) [reply Sources/PklSwift/EvaluatorManager.swift:90]: Then again, memoization can probably best be done _inside_ the Evaluator Manager... I'll roll it in.

File: Tests/PklSwiftTests/ProjectTest.swift [hunk 6]
```diff
@@ -106,7 +105,8 @@ class ProjectTest: XCTestCase {
                 modulePath: ["/bar/baz"],
                 timeout: .minutes(5),
                 moduleCacheDir: "/bar/buzz",
-                rootDir: "/buzzy"
+                rootDir: "/buzzy",
+                http: nil
```
[the remark this answers is not in the document]
Comment (holzensp) [reply Tests/PklSwiftTests/ProjectTest.swift:109]: I've made it conditional on the version.

File: Sources/PklSwift/EvaluatorManager.swift [hunk 7]
```diff
@@ -37,6 +38,57 @@ public func withEvaluatorManager<T>(_ action: (EvaluatorManager) async throws ->
...
+                continue
+            }
+            throw error
+        }
+    }
+    throw PklError("Unable to find `pkl` command on PATH.")
+}
+
+/// Get the version of the `pkl-server` being used.
+func getVersion() throws -> SemanticVersion {
```
[the remark this answers is not in the document]
Comment (holzensp) [reply Sources/PklSwift/EvaluatorManager.swift:72]: Rolled it into EvaluatorManager and let it return a `String`. Btw, it was already "internal" (as opposed to "public").

File: Sources/PklSwift/EvaluatorManager.swift (hunk 5 above)
[the remark this answers is not in the document]
Comment (bioball) [reply Sources/PklSwift/EvaluatorManager.swift]: The evaluator manager represents a child process. We don't do this yet, but at some point we should have a `newEvaluatorManagerWithCommand` constructor [like we do in go](https://pkg.go.dev/github.com/apple/pkl-go/pkl#NewProjectEvaluatorWithCommand).

For that reason, you can totally have multiple mangers, each with their own version, and that's also why the `GetVersion()` method is on the evalutor manager in Go: https://pkg.go.dev/github.com/apple/pkl-go/pkl#EvaluatorManager.

File: Tests/PklSwiftTests/ProjectTest.swift [hunk 8]
```diff
@@ -39,6 +40,20 @@ class ProjectTest: XCTestCase {
 
         let file = URL(fileURLWithPath: NSTemporaryDirectory(), isDirectory: true)
             .appendingPathComponent("PklProject")
+        let httpSetting = version < pklVersion0_26 ? "" : """
+            http {
+              proxy {
+                address = "http://proxy.mock.example.com:1"
```
Comment (bioball) [inline Tests/PklSwiftTests/ProjectTest.swift:46]: Same comment; for here and line 56: can we use `localhost` as the proxy address? `example.com` is a real domain.

**Review by bioball — APPROVED** [review]
Just one comment, but approving to unblock.

Also: same comment as before; can you run `swiftformat` on this code before committing?

**Pull request merged**